### PR TITLE
Remove Python 2.7 binary wheel generations

### PR DESCRIPTION
**Pull request by lidizheng**
Python 2.7 has been end-of-life for 1.5 yrs. This PR contains:

1. Remove the 27 artifacts;
2. Remove the "2.7 supported" label from `setup.py`.
3. Remove 27 from distribtests (centos6 is [EOL completely](https://wiki.centos.org/About/Product), so removed)
4. Remove 34 from binary distribtests (`debian:jessie` and `fedora:23`), and we still tests to build against 34 with source code

However, 27 users might still got the source wheel, e.g., `grpcio-1.38.1.tar.gz (21.4 MB)`. This is because source wheel doesn't have platform or interpretor enforcement. Technically, we can remove source wheel in `tools/run_tests/artifacts/build_artifact_python.sh`. But that will break niche users who rely on compile gRPC on their own.

---

This fixes b/191804431

### CODE REVIEW

File: tools/distrib/python/grpcio_tools/setup.py [hunk 1]
```diff
@@ -55,12 +55,7 @@
 CLASSIFIERS = [
     'Development Status :: 5 - Production/Stable',
     'Programming Language :: Python',
-    'Programming Language :: Python :: 2',
-    'Programming Language :: Python :: 2.7',
     'Programming Language :: Python :: 3',
-    'Programming Language :: Python :: 3.4',
-    'Programming Language :: Python :: 3.5',
```
Comment (gnossen) [inline tools/distrib/python/grpcio_tools/setup.py:62]: Why are we removing 3.5 and 3.6 here?

**Review by gnossen — APPROVED** [review]
Thanks for taking care of this. I know the distribtests must have been difficult to get working due to the bitrot.

File: tools/distrib/python/grpcio_tools/setup.py (hunk 1 above)
Comment (lidizheng) [reply tools/distrib/python/grpcio_tools/setup.py:62]: I found we keep forgetting to add newer Python versions to this `setup.py`. Simply state we are supporting for all non-EOL Python3 version is more accurate and less burdensome in the long run.

**Comment by lidizheng** [comment]
Known failures https://github.com/grpc/grpc/issues/26595

**Pull request merged**